### DISCUSSION TITLE
Getting 1105 error. What's causing this timeout?

### DISCUSSION THREAD

**Discussion by lsshawn** [planetscale/discussion#490] [Q&A] [answered]
`1105 (HY000): target: readablepod.-.primary: vttablet: rpc error: code = Aborted desc = transaction 1687308827047530989: ended at 2023-06-21 17:50:09.296 UTC (exceeded timeout: 20s) (CallerID: vw0sjx4q3wz2npfx35po)

`

And how can I increase the timeout?

**Comment by orware** [comment, marked as answer]
Hello @lsshawn,

We can probably better assist you on this one via https://support.planetscale.com/ so we can more easily share additional details.

However, based on a quick review on my end, since your query mentions the database name involved, I was able to see some instances of the following error related to some `INSERT` queries:

```sql
Incorrect datetime value: '2021-12-06T11:00Z' for column 'uploadDate' at row 1 (errno 1292)
```

You may have worked past those errors though already.

The short answer to your question though is that you cannot tweak something like the `max_execution_time` in our platform and transactions are limited to 20 seconds to help with general scalability/performance as longer transactions aren't recommended at larger dataset sizes. 

Depending on what you are attempting to do that is leading to the 20 second transaction timeout limit being reached we can provide more suggestions.

For example, if you are attempting an import specifically the following tip might be relevant:

> Sometimes the dump file may have everything wrapped in a START TRANSACTION; / COMMIT; transaction which will result in the import timing out if it takes more than 20 seconds to complete due to our [20 second transaction timeout limit](https://planetscale.com/docs/reference/planetscale-system-limits) so you will want to make sure those are removed prior to beginning an import.

If it's an application is resulting in the transaction timeout limit being reached, there is likely a tweak needed to help avoid/prevent that from happening but the correct one to share will depend on the context (usually the language/framework being used, whether a large loop attempting the inserts/updates is occurring within a transaction, etc.).

Hopefully this is helpful!

**Comment by isaacpeh** [comment]
Hello, I am also facing the same issue. Happens intermittently during interactions with my application. It does not wait for 20s before throwing this error and can happen to a feature as simple as logging in.

For example when logging in: (not more than 50 users in the database)
`SELECT * FROM user`

Throws the rpc error 1105 right away.

`Error: target: obs-fyp.-.primary: vttablet: rpc error: code = Aborted desc = transaction 1688589659981497971: ended at 2023-07-06 08:10:45.242 UTC (exceeded timeout: 20s) (CallerID: 4dbgreus01w00m9dqp2r)
    at PromisePoolConnection.query (C:\Users\isaac\Documents\GitHub\xxx\node_modules\mysql2\promise.js:93:22)
    at query (C:\Users\isaac\Documents\GitHub\xxx\database.js:23:31)
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {
  code: 'ER_UNKNOWN_ERROR',
  errno: 1105,
  sql: "SELECT * FROM user WHERE email = 'admin@gmail.com'",
  sqlState: 'HY000',
  sqlMessage: 'target: obs-fyp.-.primary: vttablet: rpc error: code = Aborted desc = transaction 1688589659981497971: ended at 2023-07-06 08:10:45.242 UTC (exceeded timeout: 20s) (CallerID: 4dbgreus01w00m9dqp2r)'
}`

**Reply by orware** [reply]
My main thought here is that there is a portion of the code that opens a transaction somewhere and the `SELECT * FROM user ...` query is being executed from within that transaction.

If you'd like to share more code with us for further review to understand the larger context within your application, please open up a request over at https://support.planetscale.com/ and we can continue diving into what the potential cause might be together.
